core/textile/hub/hub_auth.go

```diff
@@ -140,11 +149,26 @@ func GetHubToken(ctx context.Context, st store.Store, kc keychain.Keychain, cfg
 	}
 	log.Debug("Token Challenge: Received token successfully")
 
-	if err := storeHubToken(st, token.Value.Token); err != nil {
-		return "", err
+	_, sig, err := mbase.Decode(token.Value.Sig)
+
+	if err != nil {
+		return nil, err
 	}
 
-	return token.Value.Token, nil
+	ctx = common.NewAPIKeyContext(ctx, token.Value.Key)
+	ctx = common.NewAPISigContext(ctx, token.Value.Msg, sig)
+
+	tok := thread.Token(token.Value.Token)
+	ctx = thread.NewTokenContext(ctx, tok)
+
+	// @todo: Store appToken
+
+	// this might not be ideal, we need to store also key + msg + sig
```

Comment:
I think it's better to store it to avoid doing this slow challenge too many times